core/templates/services/user-backend-api.service.ts

```diff
@@ -0,0 +1,157 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for user data.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { UrlService } from 'services/contextual/url.service.ts';
+import { UserInfoObjectFactory, UserInfo, IUserInfoBackendDict } from
+  'domain/user/UserInfoObjectFactory.ts';
+import { AppConstants } from 'app.constants';
+import { List } from 'lodash';
+
+interface subscriptionSummary {
+  'creator_picture_data_url': string;
+  'creator_username': string;
+  'creator_impact': number;
+}
+interface IPreferencesBackendDict {
```

Comment:
We don't prefix interface names with "I" anymore. Can remove the "I" prefix? Here and everywhere else **in this file**.